src/Cli/dotnet/Commands/Tool/Run/ToolRunCommand.cs

```diff
@@ -39,4 +50,29 @@ public override int Execute()
         var result = CommandFactoryUsingResolver.Create(commandspec).Execute();
         return result.ExitCode;
     }
+
+    public CommandSpec? GetRemoteCommandSpec()
+    {
+        FilePath manifestFile = _toolManifest.FindFirst(true);
```

Comment:
Do we want a centralized manifest file for handling this?

Reply:
I dont have a strong opinion on creating a manifest file or not. I would create a folder in /temp/ to work under with a cache. @dsplaisted suggested we could just download the package without creating the manifest, but I think if it reduces the new code that needs to be created, I don't see the harm in creating this temp folder local tool manifest for the tools.